### PR TITLE
Fix/cannot use langstring entity value

### PR DESCRIPTION
**Pull request by larisaVasile**

### CODE REVIEW

File: core/libs/aspect-exporter/src/lib/rdf-list/rdf-list.service.ts [hunk 1]
```diff
@@ -80,6 +80,38 @@ export class RdfListService implements CreateEmptyRdfList, EmptyRdfList {
...
+    }
+
+    preparedList
+      .filter(list => list.created)
+      .forEach(({list, predicate}) => {
+        const listElements = this.getListElements(list);
+        const elementsToBeAdded = elements.filter(element => element.predicate.id === predicate.id).map(({literal}) => literal);
+
+        if (elementsToBeAdded.length) {
+          // this.remove(source, ...elements);
```
Comment (michelu89) [inline core/libs/aspect-exporter/src/lib/rdf-list/rdf-list.service.ts:107]: Please remove this comment, when not used.

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts [hunk 2]
```diff
@@ -42,22 +40,60 @@ export class EntityValueVisitor extends BaseVisitor<DefaultEntityValue> {
     const {aspectModelUrn} = entityValue;
     const rdfModel = this.modelService.currentRdfModel;
 
-    entityValue.properties.forEach(property => {
-      const object = this.createObjectForRDF(property);
-      const {key, value} = property;
+    const {propertyCollectionWithLangString, property} = entityValue.properties.reduce(
+      (acc, property) => (
+        isDataTypeLangString(property.key.property) && property.key.property.characteristic instanceof DefaultCollection
```
Comment (michelu89) [inline core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:45]: Is it working also for Set, SortedSet and all other list definition?

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts [hunk 3]
```diff
@@ -42,22 +40,60 @@ export class EntityValueVisitor extends BaseVisitor<DefaultEntityValue> {
...
+          : acc.property.push(property),
+        acc
+      ),
+      {
+        propertyCollectionWithLangString: [],
+        property: [],
+      }
+    );
+
+    if (propertyCollectionWithLangString.length > 0) {
```
Comment (michelu89) [inline core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:56]:
```suggestion
    if (propertyCollectionWithLangString.length) {
```

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts [hunk 4]
```diff
@@ -42,22 +40,60 @@ export class EntityValueVisitor extends BaseVisitor<DefaultEntityValue> {
...
+      const withoutEmptyLangString = propertyCollectionWithLangString.filter(prop => {
+        return prop.value.value !== null && prop.value.value !== undefined;
+      });
+
+      const langStringRdfObject = withoutEmptyLangString.map(this.createObjectForCollectionLangStringRDF.bind(this));
+      this.rdfListService.pushEntityValueLangString(entityValue, ...langStringRdfObject);
+    }
+
+    if (property.length > 0) {
```
Comment (michelu89) [inline core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:65]:
```suggestion
    if (property.length) {
```

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts (hunk 3 above)
Comment (michelu89) [reply core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:56]: You dont need (> 0)

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts (hunk 4 above)
Comment (michelu89) [reply core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:65]: You dont need (> 0)

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts [hunk 5]
```diff
@@ -42,22 +40,60 @@ export class EntityValueVisitor extends BaseVisitor<DefaultEntityValue> {
...
   }
 
-  private createObjectForRDF({key, value}: EntityValueProperty): any {
+  private createObjectForRDF({key, value}: EntityValueProperty): NamedNode | Literal {
     if (value instanceof DefaultEntityValue) {
       return DataFactory.namedNode(value.aspectModelUrn);
     }
 
-    if (isDataTypeLangString(key.property)) {
+    if (isDataTypeLangString(key.property) && key.property.characteristic instanceof DefaultCharacteristic) {
```
Comment (michelu89) [inline core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:96]: Here you make sure its not an Collection type -> List, Set, SortedSet ... remove "key.property.characteristic instanceof DefaultCharacteristic" and make sure its not an Collection Type with "! ... List, Set, SortedSet ..." Be sure you check all Collection types or that one that are extending it.

**Review by michelu89 — CHANGES_REQUESTED** [review]

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts (hunk 4 above)
Comment (larisaVasile) [reply core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:65]: done

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts (hunk 3 above)
Comment (larisaVasile) [reply core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:56]: done

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts (hunk 2 above)
Comment (larisaVasile) [reply core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:45]: done

File: core/libs/aspect-exporter/src/lib/rdf-list/rdf-list.service.ts (hunk 1 above)
Comment (larisaVasile) [reply core/libs/aspect-exporter/src/lib/rdf-list/rdf-list.service.ts:107]: done

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts [hunk 6]
```diff
@@ -42,22 +40,60 @@ export class EntityValueVisitor extends BaseVisitor<DefaultEntityValue> {
...
+        return prop.value.value !== null && prop.value.value !== undefined;
+      });
+
+      const langStringRdfObject = withoutEmptyLangString.map(this.createObjectForCollectionLangStringRDF.bind(this));
+      this.rdfListService.pushEntityValueLangString(entityValue, ...langStringRdfObject);
+    }
+
+    if (property.length > 0) {
+      const propertiesWithoutEmptyLangString = property.filter(prop => {
+        const urn = prop.key.property.characteristic.dataType.urn;
```
Comment (michelu89) [inline core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:67]:
```suggestion
        const urn = prop.key.property.characteristic?.dataType?.getUrn();
```

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts (hunk 2 above)
Comment (larisaVasile) [reply core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:45]: yes, it is working

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts [hunk 7]
```diff
@@ -71,13 +107,16 @@ export class EntityValueVisitor extends BaseVisitor<DefaultEntityValue> {
       this.setPrefix(aspectModelUrn);
     }
   }
-
+  private getDataTypeFromUrn(urn: string): string {
```
Comment (michelu89) [inline core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:110]: Remove this.

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts [hunk 8]
```diff
@@ -42,22 +40,60 @@ export class EntityValueVisitor extends BaseVisitor<DefaultEntityValue> {
...
+      });
+
+      const langStringRdfObject = withoutEmptyLangString.map(this.createObjectForCollectionLangStringRDF.bind(this));
+      this.rdfListService.pushEntityValueLangString(entityValue, ...langStringRdfObject);
+    }
+
+    if (property.length > 0) {
+      const propertiesWithoutEmptyLangString = property.filter(prop => {
+        const urn = prop.key.property.characteristic.dataType.urn;
+        return !(this.getDataTypeFromUrn(urn) === 'langString' && prop.value === '');
```
Comment (michelu89) [inline core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:68]:
```suggestion
        return !(urn === `${Samm.RDF_URI}#langString` && prop.value === '');
```

File: core/apps/ame/src/app/startup.service.ts [hunk 9]
```diff
@@ -43,7 +43,7 @@ export class StartupService {
     private fileHandlingService: FileHandlingService,
     private mxGraphService: MxGraphService,
     private translate: LanguageTranslationService,
-    private ngZone: NgZone
+    private ngZone: NgZone,
```
Comment (michelu89) [inline core/apps/ame/src/app/startup.service.ts:46]: Can you run the prettier here please.

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts (hunk 6 above)
Comment (larisaVasile) [reply core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:67]: done

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts (hunk 7 above)
Comment (larisaVasile) [reply core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:110]: done

File: core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts (hunk 8 above)
Comment (larisaVasile) [reply core/libs/aspect-exporter/src/lib/visitor/entity-value/entity-value-visitor.ts:68]: done

File: core/apps/ame/src/app/startup.service.ts (hunk 9 above)
Comment (larisaVasile) [reply core/apps/ame/src/app/startup.service.ts]: done

File: core/apps/ame-e2e/src/fixtures/entity-value/validFileText.txt [hunk 10]
```diff
@@ -0,0 +1,52 @@
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
```
Comment (michelu89) [inline core/apps/ame-e2e/src/fixtures/entity-value/validFileText.txt:1]: Add copyright

File: core/apps/ame-e2e/src/integration/export/export-entity-value-with-rdf-lang-string.cy.ts [hunk 11]
```diff
@@ -0,0 +1,52 @@
...
+      cy.visitDefault();
+      cy.fixture('/entity-value/twoCollectionSet')
+        .as('rdfString')
+        .then(rdfString => cy.loadModel(rdfString))
+        .then(() => cy.getUpdatedRDF())
+        .then(rdf => {
+          expect(rdf).to.contain(':Complaint10 a :Mode;');
+          expect(rdf).to.contain(':modeDescription ("Test"@de "Test"@en);');
+          expect(rdf).to.contain(':ModeDescription a samm-c:Collection;');
+          expect(rdf).to.contain(':ModeValue a samm-c:Collection');
```
Comment (michelu89) [inline core/apps/ame-e2e/src/integration/export/export-entity-value-with-rdf-lang-string.cy.ts:48]: missing check for 

:modeValue ( "Test"@de "Test"@en )

File: core/apps/ame-e2e/src/integration/editor/edit-property.cy.ts [hunk 12]
```diff
@@ -559,13 +552,9 @@ describe('Test edit property', () => {
...
         .then(rdf => {
-          expect(rdf).to.contain(
-            ':Entity1 a samm:Entity;\n' +
-              '    samm:properties ([\n' +
-              '  samm:property :property2;\n' +
-              '  samm:notInPayload true\n' +
-              ']).',
-          );
+          expect(rdf).to.contain(':Entity1 a samm:Entity;\n');
+          expect(rdf).to.contain('samm:property :property2');
```
Comment (michelu89) [inline core/apps/ame-e2e/src/integration/editor/edit-property.cy.ts:556]: Now we dont know if "samm:properties" contains the property2 and the notInPayload.

File: core/apps/ame-e2e/src/integration/editor/edit-property.cy.ts [hunk 13]
```diff
@@ -520,16 +520,9 @@ describe('Test edit property', () => {
...
-              '    samm:properties ([\n' +
-              '  samm:property :property1;\n' +
-              '  samm:optional true\n' +
-              ']);\n' +
-              '    samm:operations ();\n' +
-              '    samm:events ().\n',
-          );
+          expect(rdf).to.contain(':property1 a samm:Property;');
+          expect(rdf).to.contain('samm:characteristic :Characteristic1.');
+          expect(rdf).to.contain('samm:optional true');
```
Comment (michelu89) [inline core/apps/ame-e2e/src/integration/editor/edit-property.cy.ts:525]: Here we dont know if the AspectDefault contains their properties and their all the other elements.

**Review by michelu89 — CHANGES_REQUESTED** [review]

File: core/apps/ame-e2e/src/integration/export/export-entity-value-with-rdf-lang-string.cy.ts (hunk 11 above)
Comment (larisaVasile) [reply core/apps/ame-e2e/src/integration/export/export-entity-value-with-rdf-lang-string.cy.ts:48]: done

File: core/apps/ame-e2e/src/fixtures/entity-value/validFileText.txt (hunk 10 above)
Comment (larisaVasile) [reply core/apps/ame-e2e/src/fixtures/entity-value/validFileText.txt:12]: done

File: core/apps/ame-e2e/src/fixtures/entity-value/twoCollectionSet.txt [hunk 14]
```diff
@@ -0,0 +1,51 @@
+@prefix samm: <urn:samm:org.eclipse.esmf.samm:meta-model:2.1.0#> .
```
[the remark this answers is not in the document]
Comment (larisaVasile) [reply core/apps/ame-e2e/src/fixtures/entity-value/twoCollectionSet.txt:12]: done

**Review by michelu89 — APPROVED** [review]

**Pull request merged**